### PR TITLE
PageStorage: Fix entry.tag after full gc && add more debug message

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/5093, ref https://github.com/pingcap/tiflash/issues/5076

Problem Summary: The entry.tag is incorrect after full gc, this will lead to more IO caused by `RegionPersister::doPersist`
https://github.com/pingcap/tiflash/blob/5b61ae70550624d3bf0b5ca6bac89013ed5a6a4b/dbms/src/Storages/Transaction/RegionPersister.cpp#L111-L115

### What is changed and how it works?

* Copy the old entry and upodate the fields of new entry
* Add some error message when exception is thrown while reading (for #5076 )

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/BlobStore.cpp [hunk 1]
```diff
@@ -824,11 +825,20 @@ Page BlobStore::read(const PageIDAndEntryV3 & id_entry, const ReadLimiterPtr & r
...
-    blob_file->read(buffers, offset, size, read_limiter, background);
+    BlobFilePtr blob_file = getBlobFile(blob_id);
+    try
+    {
+        blob_file->read(buffers, offset, size, read_limiter, background);
+    }
+    catch (DB::Exception & e)
+    {
+        // add debug message
+        e.addMessage(fmt::format("(error while reading page data [page_id={}] [blob_id={}] [offset={}] [size={}])", page_id_v3, blob_id, offset, size));
```
Comment (jiaqizho) [inline dbms/src/Storages/Page/V3/BlobStore.cpp:839]: also add `background` pls.

**Review by jiaqizho — APPROVED** [review]
LGTM

File: dbms/src/Storages/Page/V3/BlobStore.cpp (hunk 1 above)
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/BlobStore.cpp:839]:
```suggestion
        e.addMessage(fmt::format("(error while reading page data [page_id={}] [blob_id={}] [offset={}] [size={}] [background={}])", page_id_v3, blob_id, offset, size, background));
```

**Review by hehechen — APPROVED** [review]

**Pull request merged**